### PR TITLE
CAV-217: Input validation for verify/otp

### PR DESCRIPTION
**Pull request by johno1985**
- validate phone number
 - corrected test setup, pass in unnormalised phone number as it will be normalised by the validation service

### CODE REVIEW

File: app/uk/gov/hmrc/cipphonenumberverification/connectors/ValidateConnector.scala [hunk 1]
```diff
@@ -31,10 +30,10 @@ class ValidateConnector @Inject()(httpClientV2: HttpClientV2, config: AppConfig)
 
   val validateUrl = s"${config.validateUrlProtocol}://${config.validateUrlHost}:${config.validateUrlPort}/customer-insight-platform/phone-number/validate"
 
-  def callService(phoneNumber: PhoneNumber)(implicit hc: HeaderCarrier): Future[HttpResponse] = {
+  def callService(phoneNumber: String)(implicit hc: HeaderCarrier): Future[HttpResponse] = {
```
Comment (mar36tin) [inline app/uk/gov/hmrc/cipphonenumberverification/connectors/ValidateConnector.scala:33]: why did you change this from PhoneNumber to a string

File: app/uk/gov/hmrc/cipphonenumberverification/services/PasscodeService.scala [hunk 2]
```diff
@@ -34,9 +34,9 @@ class PasscodeService @Inject()(passcodeCacheRepository: PasscodeCacheRepository
       .map(_ => phoneNumberAndOtp)
   }
 
-  def retrievePasscode(phoneNumberAndOtp: PhoneNumberAndOtp): Future[Option[PhoneNumberAndOtp]] = {
-    logger.debug(s"Retrieving phoneNumberAndOtp from database for ${phoneNumberAndOtp.phoneNumber}")
-    passcodeCacheRepository.get[PhoneNumberAndOtp](phoneNumberAndOtp.phoneNumber)(DataKey("cip-phone-number-verification"))
+  def retrievePasscode(phoneNumber: String): Future[Option[PhoneNumberAndOtp]] = {
```
Comment (mar36tin) [inline app/uk/gov/hmrc/cipphonenumberverification/services/PasscodeService.scala:37]: same here

File: app/uk/gov/hmrc/cipphonenumberverification/connectors/ValidateConnector.scala (hunk 1 above)
Comment (johno1985) [reply app/uk/gov/hmrc/cipphonenumberverification/connectors/ValidateConnector.scala:33]: Because we are now using the same function from verify/otp. verify/otp has a PhoneNumberAndOtp object so I changed the signature to take a string so that it will work for both verify and verify/otp

File: app/uk/gov/hmrc/cipphonenumberverification/services/PasscodeService.scala (hunk 2 above)
Comment (johno1985) [reply app/uk/gov/hmrc/cipphonenumberverification/services/PasscodeService.scala:37]: Because we are now using the same function from verify/otp. verify/otp has a PhoneNumberAndOtp object so I changed the signature to take a string so that it will work for both verify and verify/otp

**Review by brdgeth — APPROVED** [review]
